addons/trenches/functions/fnc_removeTrench.sqf

```diff
@@ -50,28 +56,73 @@ private _fnc_onFinish = {
     [_unit, "", 1] call EFUNC(common,doAnimation);
 };
 private _fnc_onFailure = {
-    (_this select 0) params ["_unit", "_trench"];
-    _unit setVariable [QGVAR(isDiggingId), -1, true];
-    _trench setVariable [QGVAR(digging), false, true];
+   (_this select 0) params ["_unit", "_trench"];
+   _trench setVariable [QGVAR(digging), false, true];
+   _trench setVariable [QGVAR(diggingType), nil, true];
 
-    // Save progress global
-    private _progress = _trench getVariable [QGVAR(progress), 0];
-    _trench setVariable [QGVAR(progress), _progress, true];
+   // Save progress global
+   private _progress = _trench getVariable [QGVAR(progress), 0];
+   _trench setVariable [QGVAR(progress), _progress, true];
 
-    // Reset animation
-    [_unit, "", 1] call EFUNC(common,doAnimation);
+   // Reset animation
+   [_unit, "", 1] call EFUNC(common,doAnimation);
+};
+private _fnc_condition = {
+   (_this select 0) params ["", "_trench"];
+
+   if !(_trench getVariable [QGVAR(digging), false]) exitWith {false};
+   if (_trench getVariable [QGVAR(diggerCount), 0] <= 0) exitWith {false};
+   if (GVAR(stopBuildingAtFatigueMax) && (EGVAR(advanced_fatigue,anReserve) <= 0))  exitWith {false};
+   true
 };
-[(_removeTimeLeft + 0.5), [_unit, _trench], _fnc_onFinish, _fnc_onFailure, localize LSTRING(RemovingTrench)] call EFUNC(common,progressBar);
+[[_unit, _trench, false], _fnc_onFinish, _fnc_onFailure, localize LSTRING(RemovingTrench), _fnc_condition] call FUNC(progressBar);
 
-private _progressLeft = ((1 - _actualProgress) * 10) + 1;
+[{
+  params ["_args", "_handle"];
+  _args params ["_trench", "_unit", "_removeTime", "_vecDirAndUp"];
+  private _actualProgress = _trench getVariable [QGVAR(progress), 0];
+  private _diggerCount = _trench getVariable [QGVAR(diggerCount), 0];
 
-for "_i" from _progressLeft to 10 do {
-    private _vectorDiffZ = _i / 10;
-    private _delay = _removeTime * ((_i / 10) - (1 - _actualProgress));
-    private _progress = 1 - (_i / 10);
+  if (_actualProgress <= 0) exitWith {
+     [_handle] call CBA_fnc_removePerFrameHandler;
+     _trench setVariable [QGVAR(digging), false, true];
+     _trench setVariable [QGVAR(diggerCount), 0, true];
+     deleteVehicle _trench;
+ };
 
-    [DFUNC(setTrenchPlacement), [_unit, _trench, _trenchId, _basePos vectorDiff [0, 0, _vectorDiffZ], _vecDirAndUp, _progress], _delay] call CBA_fnc_waitAndExecute;
-};
+  if (
+        !(_trench getVariable [QGVAR(digging), false]) ||
+        (_diggerCount <= 0)
+     ) exitWith {
+    [_handle] call CBA_fnc_removePerFrameHandler;
+    _trench setVariable [QGVAR(digging), false, true];
+    _trench setVariable [QGVAR(diggerCount), ((_diggerCount -1) max 0), true];
+  };
+
+  private _boundingBox = boundingBoxReal _trench;
+  _boundingBox params ["_lbfc"];                                         //_lbfc(Left Bottom Front Corner) _rtbc (Right Top Back Corner)
+  _lbfc params ["", "", "_lbfcZ"];
+
+  private _pos = (getPosWorld _trench);
```

Comment:
useless parenthesis.... Have I seen this code before? It looks alot like the other one where I found this exact same thing with the exact same useless parenthesis?

There are more in this file, won't comment them again.